substrate-node/pallets/pallet-tfgrid/src/migrations/v14.rs

```diff
@@ -0,0 +1,313 @@
+    types::StorageVersion,
+    Config, Error, FarmInfoOf, Farms, InterfaceOf, LocationOf, Nodes, PalletVersion, PubConfigOf,
+    PublicIpOf, SerialNumberOf, TFGRID_NODE_VERSION,
+};
+use frame_support::{
+    pallet_prelude::Weight, traits::ConstU32, traits::Get, traits::OnRuntimeUpgrade, BoundedVec,
+};
+use log::{debug, info};
+use sp_std::marker::PhantomData;
+use tfchain_support::{
+    traits::PublicIpModifier,
+    types::{Farm, Node},
+};
+
+use valip::ip4::{Ip as IPv4, CIDR as IPv4Cidr};
+use valip::ip6::{Ip as IPv6, CIDR as IPv6Cidr};
+
+#[cfg(feature = "try-runtime")]
+use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+pub struct FixPublicIP<T: Config>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for FixPublicIP<T>
+where
+    IP4<T>: From<<T as Config>::IP4>,
+    GW4<T>: From<<T as Config>::GW4>,
+    IP6<T>: From<<T as Config>::IP6>,
+    GW6<T>: From<<T as Config>::GW6>,
+{
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        assert!(PalletVersion::<T>::get() <= types::StorageVersion::V13Struct);
+
+        info!("current pallet version: {:?}", PalletVersion::<T>::get());
+        let nodes_count: u64 = Nodes::<T>::iter_keys().count() as u64;
+        Self::set_temp_storage(nodes_count, "pre_node_count");
+        log::info!(
+            "🔎 FixPublicIPV13 pre migration: Number of existing nodes {:?}",
+            nodes_count
+        );
+        Ok(())
+    }
+
+    fn on_runtime_upgrade() -> Weight {
+        if PalletVersion::<T>::get() == StorageVersion::V13Struct {
+            migrate_nodes::<T>() + migrate_farms::<T>()
+        } else {
+            info!(" >>> Unused migration");
+            0
+        }
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        assert!(PalletVersion::<T>::get() >= types::StorageVersion::V14);
+        // Check number of nodes against pre-check result
+        let pre_nodes_count = Self::get_temp_storage("pre_node_count").unwrap_or(0u64);
+        assert_eq!(
+            Nodes::<T>::iter().count() as u64,
+            pre_nodes_count,
+            "Number of nodes migrated does not match"
+        );
+
+        info!(
+            "👥  FixPublicIPV13 post migration: migration to {:?} passes POST migrate checks ✅",
+            PalletVersion::<T>::get()
+        );
+
+        Ok(())
+    }
+}
+
+pub fn migrate_nodes<T: Config>() -> frame_support::weights::Weight
+where
+    IP4<T>: From<<T as Config>::IP4>,
+    GW4<T>: From<<T as Config>::GW4>,
+    IP6<T>: From<<T as Config>::IP6>,
+    GW6<T>: From<<T as Config>::GW6>,
+{
+    info!(
+        " >>> Starting tfgrid pallet migration, pallet version: {:?}",
+        PalletVersion::<T>::get()
+    );
+
+    let mut migrated_count = 0;
+
+    Nodes::<T>::translate::<
+        super::types::v13::Node<LocationOf<T>, PubConfigOf<T>, InterfaceOf<T>, SerialNumberOf<T>>,
+        _,
+    >(|k, n| {
+        migrated_count += 1;
```

Comment:
I would put this line after `new_node` is created just for the logical flow
as it s done below in `migrate_farms()`